src/game-engine/GameEngineB.ts

```diff
@@ -138,10 +138,16 @@ export default class GameEngineB {
   choosePlay(bPlay: Play) {
     if (!(this.state instanceof State.ReadyToChooseBPlay)) { return this.state };
 
-    this.validateBalances();
-
     const { channel, stake, balances, preCommit, adjudicator, turnNum } = this.state;
 
+    if (this.state.stake > this.state.balances[1]) {
+      return this.transitionTo(
+        new State.InsufficientFundsB({
+          channel, balances, adjudicator,
+        })
+      )
+    }
```

Comment:
Same question as above - could we let the user know earlier?

Reply:
I think if we're going to let the user know earlier, we should commit to that, and check that both players have enough stake to cover a round. If not, A isn't allowed to choose a move.